### PR TITLE
docs/user/azure: Update Azure documention to be on par with AWS

### PR DESCRIPTION
**Pull request by jhixson74**
This is my attempt to update the Azure docs to be on par with AWS. This likely needs more work. This is a first draft. 

See: https://jira.coreos.com/browse/CORS-1172

### CODE REVIEW

File: docs/user/azure/account.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+This is what the Azure portal looks like. 
+
+![Portal](images/account_Microsoft_Azure.png)
+
+
+# Set up Azure CLI credentials
+
+To install the Linux Azure CLI tools, follow [these instructions][install-cli]. To start using the command line tools, enter:
+````console
```
Comment (abhinavdahiya) [inline docs/user/azure/account.md]: nit: new lines around the blocks.

File: docs/user/azure/account.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+This will direct you to login via a browser (which can be on another device). The CLI client will automatically retrieve and store the authz token at the end of the process. You can configure the default subscription using:
+````console
+$ az account set -s NAME_OR_ID
+````
+Collect your account info:
+````console
+$ az account show
+{
+  "environmentName": "AzureCloud",
+  "id": "4bab1c60-32d5-30b3-f68e-27a15b958a90",
```
Comment (abhinavdahiya) [inline docs/user/azure/account.md]: are these real ?? maybe zero them out.

File: docs/user/azure/account.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+Option 1: Use the Wizard
+
+First create a service principal (replacing --name with your own in the format $user-$purpose or $team-$purpuse, whichever is more appropriate):
+````console
+$ az ad sp create-for-rbac --role Owner --name team-installer
+Retrying role assignment creation: 1/36
+Retrying role assignment creation: 2/36
+Retrying role assignment creation: 3/36
+{
+  "appId": "e564ac9e-bd46-4c5a-8592-03724a82c3f2",
```
Comment (abhinavdahiya) [inline docs/user/azure/account.md]: zero them out.

File: docs/user/azure/account.md [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+  "displayName": "team-installer",
+  "name": "http://team-installer",
+  "password": "111-11-1111",
+  "tenant": "6047c7e9-b2ad-488d-a54e-dc3f6be6a7ee"
+}
+````
+When you run the installer wizard, if the credentials file doesn't exist, you'll be prompted to enter account and service principal info and the credentials file will be generated.
```
Comment (abhinavdahiya) [inline docs/user/azure/account.md]: which credential file is being referenced here.

File: docs/user/azure/account.md [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+Now that everything is filled out, you can go to the Azure portal.
+
+![Created](images/account_Welcome_to_Microsoft_Azure.png)
+
+This is what the Azure portal looks like. 
+
+![Portal](images/account_Microsoft_Azure.png)
+
+
+# Set up Azure CLI credentials
```
Comment (abhinavdahiya) [inline docs/user/azure/account.md]: also there's already a file for credentials... https://github.com/openshift/installer/blob/1f939d01fc96469c2f2205d9e71d6324b85b8a69/docs/user/azure/credentials.md

File: docs/user/azure/account.md (hunk 4 above)
Comment (abhinavdahiya) [inline docs/user/azure/account.md]: also we should provide them 2 options.

1) using the wizard..

show them the openshift-install create cluster and what values from the above must be entered when..

2) create your own cred file and provide that to the installer using `AZURE_AUTH_LOCATION` env.

File: docs/user/azure/limits.md [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+
+Each cluster creates its own VNet. The default limit of VNets per regions is 1000 and will allow 1000 clusters. To 
+have more than 1000 clusters, you will need to increase this limit. The default installation creates a single VNet.
+
+## Network Interfaces
+
+The default installation creates 6 network interfaces. The default limit per region is 65536. Additional network interfaces are
+created for additional machines and load balancers created by cluster usage and deployed workloads.
+
+## Public and Private IP Addresses
```
Comment (abhinavdahiya) [inline docs/user/azure/limits.md]: this section doesn't provide the number of public addressed used by the cluster. one for external lb, one for bootstrap-host (which is removed after installation is complete)

File: docs/user/azure/limits.md [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+By default, the installer distributes control-plane and compute machines across [all availability zones within a region][availability-zones] to provision the cluster in a highly available configuration.
+Please see [this map][az-map] for a current region map with availability zone count.
+We recommend selecting regions with 3 or more availability zones.
+You can [provide an install-config](../overview.md#multiple-invocations) to [configure](customization.md) the installer to use specific zones to override the defaults.
+
+The installer creates an internal load balancer and two subnets within the VNet. There is a subnet for the control plane nodes and there is a subnet for the compute nodes. The internal load balancer balances off the compute nodes while an external load balancer with a public IP address balances out the control plane nodes.
+
+## Network Security Groups
+
+Each cluster creates network security groups for the control plane and for compute nodes. The default limit of 5000
```
Comment (abhinavdahiya) [inline docs/user/azure/limits.md]: the network security groups in Azure are attached to subnets.

File: docs/user/azure/limits.md [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+   * 2 vCPU's, 8GiB ram
+   * IOPs / Throughput (Mbps): (cached) 4000 / 256
+   * IOPs / Throughput (Mbps): (uncached) 3200 / 384
+   * NICs / Bandwidth (Mbps): 2 / 1000
+   * 16 GiB temp storage (SSD)
+   * 4 data disks max
+
+More details on VM sizes can be found [here][sizes-general].
+
+Currently, these instance type counts are within a new account's default limit.
```
Comment (abhinavdahiya) [inline docs/user/azure/limits.md]: Azure only gives you 10 vCPUs for default subscription, might have increased.. but the default requirement of 22 vCPUs is not met by default limits.

File: docs/user/azure/account.md (hunk 1 above)
Comment (jhixson74) [reply docs/user/azure/account.md]: Fixed.

File: docs/user/azure/account.md (hunk 2 above)
Comment (jhixson74) [reply docs/user/azure/account.md]: Fixed.

File: docs/user/azure/account.md (hunk 3 above)
Comment (jhixson74) [reply docs/user/azure/account.md]: Fixed.

File: docs/user/azure/account.md (hunk 4 above)
Comment (jhixson74) [reply docs/user/azure/account.md]: Fixed.

File: docs/user/azure/limits.md (hunk 8 above)
Comment (jhixson74) [reply docs/user/azure/limits.md]: Updated (default is 20 now).

File: docs/user/azure/limits.md (hunk 7 above)
Comment (jhixson74) [reply docs/user/azure/limits.md]: Updated.

**Comment by kalexand-rh** [comment]
@jhixson74, do you have a list or permitted/suggested/preferred instance types?

**Comment by jhixson74** [comment]
> @jhixson74, do you have a list or permitted/suggested instance types?

@kalexand-rh, do you mean something like this: 

https://docs.microsoft.com/en-us/azure/virtual-machines/linux/sizes-general

?

File: docs/user/azure/account.md (hunk 4 above)
Comment (jhixson74) [reply docs/user/azure/account.md]: https://github.com/openshift/installer/blob/master/docs/user/azure/install.md already shows running openshift-install create cluster without the credentials file existing. Should I just create a link to that from this page?

File: docs/user/azure/limits.md (hunk 6 above)
Comment (jhixson74) [reply docs/user/azure/limits.md]: Updated.

File: docs/user/azure/account.md (hunk 4 above)
Comment (jhixson74) [reply docs/user/azure/account.md]: I went ahead and adding a link to the install page.

File: docs/user/azure/account.md (hunk 5 above)
Comment (jhixson74) [reply docs/user/azure/account.md]: I've removed the credentials bits from the account doc and updated the credentials page with the extra bits from the account page.

**Pull request merged**